src/ImageData.vala

```diff
@@ -0,0 +1,66 @@
+/*
+* Copyright 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+*/
+
+public class Notifications.ImageData : GLib.Object {
+    public int width { get; construct; }
+    public int height { get; construct; }
+    public int rowstride { get; construct; }
+    public bool has_alpha { get; construct; }
+    public int bits_per_sample { get; construct; }
+    public int n_channels { get; construct; }
+    public void* raw { get; set; }
+
+    public ImageData (int width, int height, int rowstride, bool has_alpha, int bits_per_sample, int n_channels, void* raw) {
+        Object (
+            width: width, 
+            height: height, 
+            rowstride: rowstride, 
+            has_alpha: has_alpha, 
+            bits_per_sample: bits_per_sample, 
+            n_channels: n_channels, 
+            raw: raw
+        );
+    }
+
+    /**
+     * Decode a raw image (iiibiiay) sent through 'hints'
+     */
+    public static ImageData from_variant(Variant img)
```

Comment:
Instead of passing the data between two constructors, you could make this a constructor like so, then just have your `Object` call in there:

```suggestion
    public ImageData.from_variant (Variant img)
```

Reply:
That's awesome, thanks for the tip. Applied this change.